File: helidon-se/src/main/java/com/github/frederikpietzko/dto/VisitDto.java
package com.github.frederikpietzko.dto;

import com.github.frederikpietzko.model.Visit;

import java.time.OffsetDateTime;

public record VisitDto(
  Long id,
  String description,
  PetDto pet,
  OffsetDateTime dateTime
) {

  public static VisitDto fromEntity(Visit entity) {
    return new VisitDto(
      entity.id(),
      entity.description(),
      PetDto.fromEntity(entity.pet()),
      entity.dateTime()
    );
  }
}

File: javalin/src/main/kotlin/com/github/frederikpietzko/dto/VisitDto.kt
package com.github.frederikpietzko.dto

import java.time.OffsetDateTime

data class VisitDto(
  val id: Long,
  val description: String,
  val pet: PetDto,
  val dateTime: OffsetDateTime,
)

fun com.github.frederikpietzko.model.Visit.toDto() = VisitDto(
  id = id!!,
  description = description,
  pet = pet.toDto(),
  dateTime = dateTime,
)
